### ISSUE TITLE
How do I get print output from PrintShop?

### ISSUE THREAD

**Issue by Keatah** [AppleWin/AppleWin#385]
How do I get print output from PrintShop? Can it somehow be sent to a PDF like how Virtual II does it?
http://www.virtualii.com/

**Comment by tomcw** [comment]
No sorry, there's currently no print to PDF (or any support for graphical printing) from AppleWin.
All I can suggest is PrintScreen to capture the AppleWin screen to a bitmap file, then print that. Not what you want, I know :-/

**Comment by MarcoVerpelli** [comment]
Some time ago I tried to get b/w prints using what can be found in this thread:
https://groups.google.com/forum/#!searchin/comp.sys.apple2/imagewriter%7Csort:relevance/comp.sys.apple2/emByCev_-20/syNX59uQckQJ
As you can see a (virtual) needle is continuously fired.
![mouse](https://user-images.githubusercontent.com/28804429/28410821-27adccaa-6d3f-11e7-91e9-a8ab5a775758.png)
Unfortunately I know anything about PostScript, maybe for some expert this can be a starting point.

**Comment by sicklittlemonkey** [comment]
Can you upload the PostScript file somewhere?​

The AppleWin printer still strips the high bit, but that shouldn't matter
for PS in text mode. And in binary PS it would cause complete failure.

Cheers,
Nick.

**Comment by MarcoVerpelli** [comment]
Here is it:

https://drive.google.com/file/d/0Bz0ReUsiwYcpalpOZ21mQ09sQm8/view?usp=sharing

**Comment by sicklittlemonkey** [comment]
Thanks Marco.​

I see how this works. PostScript is powerful ... way, way too powerful. ; -
)

If you captured the printer data from AppleWin then the problem is
definitely that AppleWin strips the high bit.
Look in the source code for ParallelPrinter, there is a complicated output
routine. For an Apple II it does c & 0x7F.

We could add yet another checkbox option for this (SIGH!) or ... Hmm, I
need to have a think ...

Cheers,
Nick.

**Comment by MarcoVerpelli** [comment]
Yes I have it captured from AppleWin. Encoding & Filter turned off.
I know that the PS file was intended for "text" output from AppleWorks, and it works great!
But why not give it a try with "graphic" data?

**Comment by MarcoVerpelli** [comment]
I used MousePaint 1.2 and the pic is on the disk, Quite obviously I use the included ImageWriter driver, not my homebrew driver for EPSON LX-8000.

https://drive.google.com/open?id=0Bz0ReUsiwYcpS0dtMWZ2Smt2MzA

**Comment by sicklittlemonkey** [comment]
Thanks.

Yes, this is a problem with AppleWin's printer support. It assumes text-only, and so it strips the high-bit for all output. The Apple II Parallel Card is very simple. If we switched to another card it might support 7 and 8-bit modes in software, but for now I think the easiest thing is to change the meaning of "Filter unprintable characters". This should enforce the existing text mode. If disabled then binary would possible as in this PDF.
[Mouse.pdf](https://github.com/AppleWin/AppleWin/files/1165262/Mouse.pdf)

Cheers,
Nick.

**Comment by MarcoVerpelli** [comment]
Nearly perfect! I still see some thin white lines, but it is a great improvement.
Graphis output from AppleWin is in the home stretch.

**Comment by Keatah** [comment]
Wow. From a question asked - to a whole new feature. That's nice.

**Comment by sicklittlemonkey** [comment]
The ImageWriter emulator (written in PostScript) has been around for quite
a while apparently. I'd just never heard of it.

Cheers,
Nick.

**Comment by tomcw** [comment]
Hi Nick - can you share a link to the ImageWriter emulator?

**Comment by sicklittlemonkey** [comment]
Sure, but it's linked to in the group and also in Marco's PS file. Basically you just append the (8-bit) printer data to this PS header:
[ImageWriterEmulator-NoLF.ps.txt](https://github.com/AppleWin/AppleWin/files/1168047/ImageWriterEmulator-NoLF.ps.txt)
(I had to disable the CR+LF? option.)

There are any number of ways to append the data, but be careful in a text editor because 8-bit data may be detected as UTF-8 or some funky codepage on Windows, which when saved will corrupt the file. I joined the files in CMD.EXE with:
copy /b ImageWriterEmulator-NoLF.ps+Mouse.raw Mouse.ps

Then view and convert to PDF with GSView:
http://www.gsview.com/
(Ghostscript has more recent releases, but GSView is easier to use.)

As the size of the interpreter shows (although PS is literally built for this) it really wouldn't be that hard to emulate the ImageWriter or an Epson FX-80 internally to AppleWin. Free time is the limiting factor...

Cheers,
Nick.
